lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -875,7 +875,11 @@ def validateFASTQ(self, ctx, params):
                 t.flush()
                 shutil.copy2(t.name, file_path)
             validated = 1
-            if c % 4 != 0:
+            if p.get('interleaved') and c % 8 != 0:
```

Comment:
This needs a test

Reply:
I did test all permutations on CI. I will see if I can figure out how to add a test in for it.  We also did not have a test for the %4 before, from what I can tell.

Reply:
This might be it: https://github.com/kbaseapps/ReadsUtils/blob/master/test/ReadsUtils_server_test.py#L638

Reply:
That being said, this whole module needs a rewrite. It was the first SDK module ever AFAIR and my code quality was not high.

Reply:
Ok I added a test.